app/src/main/java/org/oppia/app/player/state/answerhandling/InteractionAnswerHandler.kt

```diff
@@ -14,6 +14,16 @@ interface InteractionAnswerHandler {
    */
   fun isExplicitAnswerSubmissionRequired(): Boolean = true
 
+  /** Return the current answer error messages. */
+  fun getPendingAnswerError(): String?{
+    return null
+  }
+
+  /** Return the current answer error messages on submit button click. */
+  fun getPendingAnswerErrorOnSubmit(): String? {
```

Comment:
Why do we need this? The submit button is handled outside of the view model (it should be handled within the state fragment presenter). I would expect calling getPendingAnswerError() upon submit should be clearer.

Unless the point here is to distinguish between real-time and submission time errors. If that's the case, let's alter getPendingAnswerError add an enum here along the lines of:

```kotlin
fun getPendingAnswerError(category: AnswerErrorCategory): String? = null

/** Categories of errors that can be inferred from a pending answer. */
enum AnswerErrorCategory {
  /** Corresponds to errors that may be found while the user is trying to input an answer. */
  REAL_TIME,
  /** Corresponds to errors that may be found only when a user tries to submit an answer. */
  SUBMIT_TIME
}
```

These seem pretty general purpose and hopefully versatile enough to be usable in all interactions.

Reply:
Done